src/core/server/rendering/rendering_service.tsx

```diff
@@ -117,4 +133,17 @@ export class RenderingService {
 
     return ((await browserConfig?.pipe(take(1)).toPromise()) ?? {}) as Record<string, any>;
   }
+
+  async checkURLValid(url: string, logger: LoggerFactory): Promise<string> {
```

Comment:
can this also verify that the url is contains 'svg' or 'png'. Something like "https://stackoverflow.com/questions/9714525/javascript-image-url-verify". I passed an actual website that succeeds this but it's not an image and it crashes the site without any clear error.

Reply:
can logger be assigned to the current context of the service and thus you can reference it by 'this.logger'

Reply:
Do we only want to limit the URL as svg and png? Do we allow other image format such as BMP, JPG? 

Also i think in most cases, the image URL will end with PNG, SVG etc. But some image URLs may go with a web page URL, and do we want to consider that? Or we should explicitly ask user to input image URL ending with PNG, SVG?

Reply:
PNG and SVG are the better image formats, since they can handle transparency. Of these, SVG is preferred, since it also allows for clean resizing without quality loss. I would prefer not to support JPG, BMP, GIF

Reply:
Agree. We will just allow SVG and PNG for URL.